cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -229,6 +233,18 @@ public ScheduleState scheduleState(ProgramId program, SchedulableProgramType pro
     }
   }
 
+  @Nullable
+  public static String removeAppVersion(String scheduleId) {
+    String[] parts = scheduleId.split(":");
+    if (parts.length != 6) {
+      return null;
+    }
+
+    List<String> stringParts = new ArrayList<>(Arrays.asList(parts));
+    stringParts.remove(3);
```

Comment:
Fixed in this commit https://github.com/caskdata/cdap/pull/8486/commits/ee29e842481d28375712440b2191e0e14fb4e129